src/onegov/api/path.py

```diff
@@ -37,10 +39,12 @@ def get_api_endpoint(
     if endpoint == 'authenticate':
         return AuthEndpoint(app)
 
-    cls = ApiEndpointCollection(app).endpoints.get(endpoint)
-    if not cls:
+    item = ApiEndpointCollection(request, app).endpoints.get(endpoint)
+    if not item:
         raise ApiException('Not found', status_code=404)
-    return cls(request, extra_parameters=extra_parameters, page=page)
+    item.page = page
+    item.extra_parameters = extra_parameters or {}
```

Comment:
I'm also not super happy about this, although the way we use it currently it seems harmless.

My preference would be something along the lines of:
- Keep the `endpoints` `cached_property`
- Add a new method `get_endpoint(self, name: str, page: int = 0, extra_parameters: dict[str, Any] | None = None) -> ApiEndpoint[Any] | None`
- Extend `config.setting_registry.api.endpoints` with the same optional parameters
- Use `get_endpoint` here instead of `endpoints.get`

This seems a little less messy and error-prone to me. Changing attributes on cached properties is always a little bit dangerous as soon as there is some concurrency or parallelism with a shared collection.